packages/angular-workspace/nimble-angular/src/directives/select/tests/nimble-select-control-value-accessor.directive.spec.ts

```diff
@@ -336,4 +337,158 @@ describe('Nimble select control value accessor', () => {
+
+            @Input() public selectValue = '';
+
+            public selectOptions: { name: string, value: number }[] = [
+                { name: 'Option 1', value: 1 },
+                { name: 'Option 2', value: 2 },
+                { name: 'Option 3', value: 3 }
+            ];
+
+            public selectedOption = new FormControl<{ name: string, value: number } | null>(null);
+
+            public form: FormGroup = new FormGroup({
+                selectValue: this.selectedOption
+            });
+
+            public get selectValueField(): AbstractControl {
+                return this.form.get('selectValue')!;
+            }
+
+            public ngOnInit(): void {
+                this.selectOptions.push({ name: 'Option 4', value: 4 });
+                this.selectValueField.setValue(this.selectOptions[3]);
+            }
+        }
+
+        let select: Select;
+        let fixture: ComponentFixture<TestHostComponent>;
+        let testHostComponent: TestHostComponent;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleSelectModule, NimbleListOptionModule, NimbleListOptionGroupModule, FormsModule, ReactiveFormsModule]
+            });
+        });
+
+        beforeEach(async () => {
+            fixture = TestBed.createComponent(TestHostComponent);
+            testHostComponent = fixture.componentInstance;
+            select = testHostComponent.select.nativeElement;
+            fixture.detectChanges();
+            // wait for select's 'options' property to be updated from slotted content
+            await waitForUpdatesAsync();
+        });
+
+        afterEach(() => {
+            processUpdates();
+        });
+
+        it('can set value to option that is dynamically added during init', () => {
+            expect(select.selectedIndex).toBe(3);
+        });
+    });
+
+    describe('dynamically added options to group on update', () => {
+        @Component({
+            template: `
+                <form [formGroup]='form'>
+                    <nimble-select #select formControlName="selectValue">
+                        <nimble-list-option-group label="Group 1">
+                            <nimble-list-option *ngFor="let option of selectOptions"
+                                [ngValue]="option">
+                                {{ option.name }}
+                            </nimble-list-option>
+                        </nimble-list-option-group>
+                    </nimble-select>
+                </form>
+             `
+        })
+        class TestHostComponent implements OnChanges {
+            @ViewChild('select', { static: true }) public select: ElementRef<Select>;
+
+            @Input() public selectValue?: { name: string, value: number };
+
+            public selectOptions: { name: string, value: number }[] = [
```

Comment:
prefer as const

Reply:
I modify the array below, so it doesn't allow `as const`. I've removed the redundant typing declaration however.